app/src/main/java/com/duckduckgo/app/cta/ui/CtaViewModel.kt

```diff
@@ -0,0 +1,193 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.cta.ui
+
+import android.view.View
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import androidx.annotation.WorkerThread
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.duckduckgo.app.browser.R
+import com.duckduckgo.app.cta.db.DismissedCtaDao
+import com.duckduckgo.app.cta.model.CtaId
+import com.duckduckgo.app.cta.model.DismissedCta
+import com.duckduckgo.app.cta.ui.CtaConfiguration.*
+import com.duckduckgo.app.feedback.db.SurveyDao
+import com.duckduckgo.app.feedback.model.Survey
+import com.duckduckgo.app.global.install.AppInstallStore
+import com.duckduckgo.app.global.install.daysInstalled
+import com.duckduckgo.app.statistics.VariantManager
+import com.duckduckgo.app.statistics.VariantManager.VariantFeature.AddWidgetCta
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.app.statistics.pixels.Pixel.PixelName.*
+import com.duckduckgo.app.widget.ui.WidgetCapabilities
+import io.reactivex.schedulers.Schedulers
+import kotlinx.android.synthetic.main.include_cta_buttons.view.*
+import kotlinx.android.synthetic.main.include_cta_content.view.*
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class CtaViewModel @Inject constructor(
+    private val appInstallStore: AppInstallStore,
+    private val pixel: Pixel,
+    private val surveyDao: SurveyDao,
+    private val widgetCapabilities: WidgetCapabilities,
+    private val dismissedCtaDao: DismissedCtaDao,
+    private val variantManager: VariantManager
+) {
+
+    val ctaViewState: MutableLiveData<CtaViewState> = MutableLiveData()
+    val surveyLiveData: LiveData<Survey> = surveyDao.getLiveScheduled()
+    private var activeSurvey: Survey? = null
+
+    data class CtaViewState(
+        val cta: CtaConfiguration? = null
+    )
+
+    fun onSurveyChanged(survey: Survey?) {
+        activeSurvey = survey
+        refreshCta()
+    }
+
+    fun refreshCta() {
+        surveyCta()?.let {
+            ctaViewState.postValue(ctaViewState.value!!.copy(cta = it))
+            return
+        }
+
+        Schedulers.io().scheduleDirect {
+            if (canShowWidgetCta()) {
+                val ctaType = if (widgetCapabilities.supportsAutomaticWidgetAdd) AddWidgetAuto else AddWidgetInstructions
+                ctaViewState.postValue(ctaViewState.value!!.copy(cta = ctaType))
```

Comment:
instead of lots of `ctaViewState.value!!` or `ctaViewState.value?` we could rely on it not being null since we initialize it in `init` i think

for convenience, we could add a function to obtain the current view state, enforcing that it's not null (we do this elsewhere, like `BrowserTabViewModel` for instance)

e.g.,

    private fun currentLoadingViewState(): LoadingViewState = loadingViewState.value!!

Reply:
Good idea. I've used a computed property:
```
private val ctaViewStateValue: CtaViewState
  get() = ctaViewState.value!!
```

Reply:
Can we just call it `currentViewState`?